include/zephyr/bluetooth/addr.h

```diff
@@ -54,16 +54,20 @@ typedef struct {
 	bt_addr_t a;
 } bt_addr_le_t;
 
+/* Global Bluetooth address constants defined in bluetooth/common/addr.c */
+extern const bt_addr_t bt_addr_any;
+extern const bt_addr_t bt_addr_none;
+extern const bt_addr_le_t bt_addr_le_any;
+extern const bt_addr_le_t bt_addr_le_none;
```

Comment:
We probably should add this to the release notes, as we are now declaring `extern` variables with these specific names, and it's not impossible that someone out there have local variables with these names. 

Alternatively we can add a `_` prefix (e.g. `_bt_addr_any`) to indicate that these shouldn't be used directly. 

Yet another alternative is to have `BT_ADDR_ANY` be a function that returns the pointer to `bt_addr_any` to avoid these `extern` variables altogether .

Reply:
We already have many global variables like this in the codebase and this is nothing out of ordinary, so I do not see why this needs to be treated specially.

As for prefixing, it is already prefixed `bt_`, which is sufficient IMHO. Downstream users should not be defining any symbols prefixed `bt_` because it is used by the Zephyr Bluetooth subsystem. This is the same for any other global names we have in our codebase (e.g. `k_`, `can_`, `uart_`, ...).

Reply:
We already own the `bt_*` namespace, so I don't think anything is needed here. We don't have any additional requirements when we want to add new `bt_*` symbols. Doesn't matter if it's function names, struct names, or such variables. Any of these can cause conflicts for an application that has decided to use `bt_*` for its own symbols.

Reply:
Fair, I see that we do have roughly 100 or so global variables treated this way. It isn't something I would consider at good design, but there's no point in treating this specially.